### PR TITLE
Remove chatty "waiting for pod" msg from kubectl run

### PR DESCRIPTION
**Pull request by duglin**
Attacking #28695 one step at a time

Signed-off-by: Doug Davis <dug@us.ibm.com>

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37164)

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

**Comment by duglin** [comment]
@k8s-bot gci gke e2e test this

**Comment by dims** [comment]
@k8s-bot gci gke e2e test this

**Comment by duglin** [comment]
@dims thanks

File: pkg/kubectl/cmd/run.go [hunk 1]
```diff
@@ -384,7 +384,7 @@ func contains(resourcesList map[string]*unversioned.APIResourceList, resource un
 
 // waitForPod watches the given pod until the exitCondition is true. Each two seconds
 // the tick function is called e.g. for progress output.
```
Comment (MHBauer) [inline pkg/kubectl/cmd/run.go]: update docs about tick?

**Comment by duglin** [comment]
@dims could you kick the tests on this one?

**Comment by dims** [comment]
@k8s-bot gce etcd3 e2e test this
@k8s-bot kops aws e2e test thi

**Comment by duglin** [comment]
ping @pwittrock maybe?

**Comment by pwittrock** [comment]
@AdoHe Will you be able to review this?

**Comment by AdoHe** [comment]
@duglin generally LGTM, could you please paste the output here? for example the output of:
```
kubectl run test --rm --restart=Never -it --image=ubuntu -- date
```
thanks very much.

**Comment by duglin** [comment]
@AdoHe sure.  Here's the output before the PR:
```
$ kubectl run test --rm --restart=Never -it --image=ubuntu -- date
Waiting for pod default/test to be running, status is Pending, pod ready: false
pod "test" deleted
```

And the output after this PR:
```
$ kubectl run test --rm --restart=Never -it --image=ubuntu -- date
pod "test" deleted
```

Now, notice the "date" output is missing - that's another issue I'm hoping we fix as part of the work on https://github.com/kubernetes/kubernetes/issues/28695 but that's not part of this PR. The difference in the output on this PR is the missing "Waiting for..." line of text.

We'll also address the chatty "pod xxx deleted" stuff in another PR.  Baby steps  :-)

**Comment by duglin** [comment]
Updated previous comment with correct output - I think my base master was a bit old. Now the 'date' output **is** shown for me

**Comment by AdoHe** [comment]
```
We'll also address the chatty "pod xxx deleted" stuff in another PR. Baby steps :-)
```
That's fine, would you like to help with too?

**Comment by duglin** [comment]
@AdoHe the other "chatty" PR is here:  https://github.com/kubernetes/kubernetes/pull/38429

**Pull request merged**